T H I S    I S    A    T O - D O    L I S T



* Figures
** TODO Read breakdown before and after curation of psol (curation.r)
*** Recalculate read counts with maximum number of reads...
** Comparative genomics of C. beij (+ others)
** 

* Why so low in Refcds??
** Reads with failed quality
** Unassembled data
** Assembled data with partial or multiple alignments to genome
** Background signal

* TODO Graphs and Statistics for assembly/annotation
** Reads (total and per sample)
*** 570,460,211 rRNA free reads
*** mapped reads
**** total mapped 562,382,753
**** both mapped 548,387,224
**** singleton read (unpaired): 8,960,839 + 5,034,690 = 13,995,529
**** properly paired 458,814,860 

** Distributions
*** Feature size
**** Transcript length (novel vs reference containing vs total)
**** CDS length (novel vs reference)
**** Operon length (assembly vs paredes)
*** Operon size
**** Paredes vs assembly
*** UTR length
**** 5' 3' UTRs
**** Intra-operonic UTRs (assembly vs paredes)
*** IGR
**** Reference vs paredes vs reannotation vs assembly
*** Coverage
**** Percentiles of transcript
**** Across first/last 100bp of transcript
**** Carlos vs assembly
** Percentages
*** Assembled basepairs/Megabases 
**** Novel 
**** Standard 
**** CDS 
**** Total
**** awk '{sum += $4/5} END {print sum}'
*** Number of transcripts
**** Novel
**** Standard
**** CDS
**** Total
*** Expressed basepairs
**** Per sample (vs entire genome)


* Shawn
** Sig F, E, G
** Shawn's thesis on shared drive
* Assembly
** METHODS
*** Realignment with blast/blat/bwa
*** Sortblast.rb
**** Separates transcripts with unique, partial, or multiple alignments
*** Only use 'unique' transcripts
*** Then use assemblycompare.rb
** TODO Remove misassembled transcripts
*** Expression thresholds
**** >= 10 reads per sample
**** >= 500 reads total
** TODO Split fused transcripts
*** Flow over transcript (start+20 -> end-20)
*** run KS test for each base
** TODO Address spurious antisense detection
*** Spearman Correlation
*** Anticorrelation over time
*** Expression thresholds
** TODO Multiple transcription start site detection
*** acquire the utr/5' region of transcript
*** cluster the bases by coverage
*** Statistically assess 5' most base / TSS
*** If gene is differentially expressed, test whether there is differential coverage
* Annotation REMEMBER ALTERNATIVE!
** Alternative for novel transcripts
*** Use ORF finder that produces a gtf/bed
*** Intersect these ORFs with the transcript set
*** Then set difference the 'standard' cdses
** METHOD
*** Realign assembled transcripts to genome
*** Utilize only 'unique' matches (<30bp in gaps, single optimal alignment)
*** extract the genomic sequences of the aligned regions only
*** Use the ORF finder Transdecoder to predict ORFs (both trained/untrained)
*** Combine and process results, converting to genomic coordinates, formating
*** Convert the annotation to genbank
*** Submit to RAST for annotation and comparison
** Transdecoder (NO TRAINING)
*** 1. Bedtools to find novel CDS
*** 2. Add novel CDS to standard cds
*** 3. Format crap for assemblyconvert
**** grep -v 'complete' results.transdecoder.cds | grep '>' | ruby -ne 'line=$_.chomp; line=line.split[0][1..-1] > results.incomplete.cds
**** sed -i 's/|/~/' results.transdecoder.cds
**** mv results.transdecoder.gff3 results.transdecoder.all.gff3
**** grep -v 'partial' results.transdecoder.gff3 | grep -v 'internal' > results.transdecoder.gff3
**** ruby scripting to import 'incomplete' cdses from above, remove from fasta records (mapping the keys back ~ -> | )
**** and also from cds records of the gff3
*** 4. assmeblyconvert.rb
**** changes the transcript and cds fastas, the transcript and cds gtfs into genomic coordinates,
**** parses to genbank
** Notes
*** 185301 | 32 (genome reannotation)
**** vs. regular annotation (+ 105)
***** Signaling
****** ABC transport/protein membrane-translocation
****** Rad50 DNA-double strand break repair
****** hyperoxide resistance regulator
****** Flagellar motility chemotaxis
****** Toxin/antitoxin system
****** Oxidative stress regulators
****** Antibiotic resistance/exporters
***** Metabolism
****** 2 extra genes in TCA (fumarate dehydrogenase,PEP carboxykinase)
****** 2 extra genes in TCA (3-OA synthase [FabH,FabF])
****** 1 extra gene in lysine biosynthesis (N-Ac-diaminopimelate deacetylase)
****** 1 extra gene in nicotinate biosynthesis (Alkaline phosphodiesterase)

*** 191494 | 33 (transdecoder only)
**** vs reannotation (+2, 1359 in common, -709)
*** 191496 | 35 (standard only)
*** 191495 | 34 (merged)
**** vs reannotation (+72 , 1426 in common, -154)
***** Added
****** Formate efflux transporter
****** Rad50
****** ABC transport
****** Magnesium and cobalst efflux CorC
****** DNA repair methylase
****** Acid-soluble sport protein SASP
****** Spor germination SleL
****** Secretion systems
****** Nitate/nitrite sensor
****** protease
****** nucleotide processing (signaling/secondary messenger)
****** Proteint processing/aminopeptidases
****** Fe hydrogenase
****** betaine/osmotic transport systems
****** glutaredoxin/glutathione hydrolases (osmotic stresss)
****** ABC sulfate transport
****** HSP33
****** Antibiotic resistance exporters/permeases (tetr, mfs, macB
****** Two component system regulator BceR
****** Copper homeostatis
****** Vancomycin resistance regulator
***** removed
****** 5-10-methyl-THF reductase
****** folate biosynthesis kinases
****** RecJ and DNA repair systems
*** 192316 | 36 (untrained)
*** 

* LATER
** Expression Analyses
*** D.E
*** Ontology/enrichment
**** Methods
***** First, map CAC to GeneIDs through 'extract.rb'
***** THEN DAVID!!
***** If not:
****** map GeneID -> Uniprot KB ID (select all, download 'target list')
******  -> DAVID
**** Subsets
***** Factors:
****** By time point
****** By stress
****** By up/down-regulation
**** Visualizations
***** Circos map
****** Add up the silhouettes/davies/dunn
****** Divide total chromosome size by sum, call 'size'
****** Each 'unit' of the metric is worth 1 'size'
****** For each cluster, start end, based on value collate:
chr,start,end,metric,svgcluster=x,svggo=blahblahblah
***** D3 Interaction matrix
****** Sort by clustering
*** Expression clustering
**** Supplementary
***** Preprocessing
****** Data normalization [[http://manuals.bioinformatics.ucr.edu/home/R_BioCondManual#TOC-Clustering-and-Data-Mining-in-R][Scaling]] (DE normalization??)
****** Regularization? (a la DEseq?
***** Feature selection
****** Differential genes only
***** Distance metric
****** Distance matrix selection (e.g. Minkowski measures: Euclidean, Manhattan, Chebyshev OR Mahalanobis**)
******* Useful when searching for EXACT matches between profiles
****** Correlation matrix ***
****** Conditional entropy
***** Linkage
****** Single (nearest)
****** Complete (farthest)
****** Average
****** Centroid (distance between centroid of each cluster)
****** Ward (minimizes ANOVA SSE between the clusters)
***** THEN: CLUSTER [[http://cran.at.r-project.org/web/views/Cluster.html][R portal to clustering]]
***** Considerations
****** Partical or complete clustering? (Do all genes need a cluster?)
***** Techniques
****** Hierarchical clustering (agglomerative)
******* Assumptions:
******** None, well that the criterion for calculating (a)similarity is appropriate
******* Options:
******** Connectivity constraints
******* Pros:
******** Easy to implement
******** Informative
******* Cons:
******** Debate over utility: pergormance is close to random
****** K-means (Partitive; iterative)
******* Assumptions:
******** Clusters are convex
******* K-medians
******* K-medoids (PAM)
******* Spectral clustering
******* Tight clustering (random resampling)
******** Noise tolerant
******* MGKM (higher cost, slight improvement)
******* IGKA (incremental genetic K-means algo.; combines genetic algorithm and K-means)
******** Uses 'Total within cluster variance' as fitness function
******** Incrementally clusters centroids
******** Sensitive to by GA parameters (mutation rate, generation, etc.)
****** Density based clustering
******* Pros/cons
******** Pros
********* No assumptions on convexity
******** Cons
********* Non-deterministic
******* EnDBSCAN/OPTICS
******* Mean-shift (uses KDE)
****** Self organizing maps (Neural nets)
******* SOTA
******* DGSOT
******* GHTSOM
******** Very unique training criteria
******** New nodes are trained only using input which caused the parent node to fire
******** Inputs mapping to leaf neurons at the top of the hierarchy are usually noise!!
****** Biclustering
******* FLOC (George Church??)
******* Possibilistic biclustering
******** H score
****** Fuzzy clustering
******* Fuzzy C-means/ FLAME
******** Differ by weighting; A gene's FCM membership score is proportional to similarity to cluster mean)
******** FLAME determined by weighted similary to K-nearest neighbors and their membership score to cluster

****** Search based
******* Simulated annealing
******* FLOC
******* Expectation maximization
****** Graph theoretic
******* CAST
***** Evaluation
****** fpc
******* Methods:
******** dbscan
******** mergenormals (gaussian mixture, use with centered, regularized)
******** kmeansruns
******** pamk
******* Resampling:
******** Clusterboot
******* prediction.strength (compute optimal number of clusters)
******* (plotcluster) graphing of a clustering
****** clValid (GO and other metrics@!!)
****** clv
****** clusplot (cluster package)
****** pvclust (hierarchical restrapping -> p-values; remember to transpose)
** Promoter motifs
** sRNA targeting
* Visualizations
** TODO Main graph
*** Add cluster visualization
*** Show tooltip
**** Implementation
***** Separate file with gene expression matrix (30 columns x 4k rows)
***** Load file in before hand, plot all points
***** On mouseover:
****** Dim other points
****** Plot 3 lines, each with 10 points
******* [[http://www.d3noob.org/2013/01/smoothing-out-lines-in-d3js.html]['Basis' interpolation]]
****** 4 time points: 15m x2pts, 75m x3pts, 150m x2pts, 270 x3pts
****** Plot [[http://bl.ocks.org/ndarville/6552457][confidence intervals]]
***** On mouseout
***** remove svg? or hide element
** Glyph graph for Terry
*** Kegg browser with glyph plot for stress, strand, time
*** Show transcripts (orfs inside)
*** Show glyphs/circles above/below each transcript (according to strand)
*** glyph fill intensity/saturation and area are proportional to expression

** Other d3 graphs:
*** Scatterplots
**** MA plots
***** TODO CSV<- gene name, expression value, log fold change, pvalue, and comparison
***** Make index.html and others
**** Correlation matrices
*** Heatmaps
**** With clustering...
** TODO Visualization with d3
*** Differential expression !!!!
**** Scatterplot
***** [[http://www.nytimes.com/interactive/2012/05/17/business/dealbook/how-the-facebook-offering-compares.html?_r=0][Scatterplot with labels]]
***** AND THIS ONE [[http://bl.ocks.org/mbostock/4063663][scatteplot with brushing]]
***** View the gene name, differential expression condition, pvalue others
**** Linkages: GO term with genes
***** [[http://www.findtheconversation.com/concept-map][concept map]]
***** Reorganizing network map [[http://blog.nextgenetics.net/demo/entry0019/demo.html][GO terms]]
***** x DE genes in the center
***** linked to GO terms on the outside
***** Each gene is linked to a condition/time combination? show expression level, percentage changes?
***** Allows more detailed exploration of the gene
***** Each gene ontology is linked to a graph which describes the enrichment of the gene ontology term in certain conditions

**** Interactive heatmap
***** [[http://bl.ocks.org/ianyfchang/8119685][Interactive heatmap]]
***** Ordering by cluster?? [[http://blog.nextgenetics.net/demo/entry0044/][condition comparison]]
***** Display fold change?
*** Coexpression clustering with d3
**** Circular clustering and interaction
***** Demo code: [[https://mbostock.github.io/d3/talk/20111116/bundle.html][Plain html: view source]]
***** BETTER SOURCE: https://gist.github.com/mbostock/1044242
***** Each record of json database has 'imports' a dictionary of linkages
***** Each import becomes colored red when hovering over a particular gene.
***** But, Each record that imports that gene is colored in green
***** USE Projects/d3/hierarchical_edge_bundling_interactive/bundle.html
***** Load data into json database
**** Circular clustering with magnitude
***** May be useful for genes or for gene ontology terms
***** [[http://bost.ocks.org/mike/uberdata/][Circular plot]]
**** Hive plot clusterin and interaction
***** May be harder to use... [[http://bost.ocks.org/mike/hive/][hive plot]]
**** Heatmap: clustering, [correlation magnitude, p-value?] - gene vs GO term
*** Gene ontology visualizations
**** Node link tree: [[http://bl.ocks.org/mbostock/4063550][Reingold-Tilford algorithm]]
**** Circular interaction with magnitudes [[http://bost.ocks.org/mike/uberdata/][Circular interaction]]
**** DITTO: [[http://redotheweb.com/DependencyWheel/][Dependency wheel]]
**** Genes vs gene ontology terms - co-occurence matrix: [[http://bost.ocks.org/mike/miserables/][Les Mis co-occurrence]]
**** 
*** Correlation matrices
**** [[https://mbostock.github.io/d3/talk/20111116/iris-splom.html][interactive subsetting]]
**** For differential expression viewing
**** Another version (older): [[http://benjiec.github.io/scatter-matrix/demo/demo.html][older]]
*** Sankey (non-circular) Flow diagrams with magnitude
**** May be useful for visualizing the change in gene ontology over time...
**** [[http://bost.ocks.org/mike/sankey/][Sankey diagrams]]
**** Can this be modified to display the number associated with the magnitude?
**** As alluvial diagram
*** Sunburst Flow diagrams with percentages
**** Very exhaustive... may visualize something interesting. or not.
**** [[http://bl.ocks.org/kerryrodden/7090426][Sequence sunburst]]
*** Slider subsetter with histograms
**** [[http://square.github.io/crossfilter/][Crossfilter]]
*** Genomic viewer for coverage
**** [[http://square.github.io/crossfilter/][Cubism]]
*** As an illustration to show a eulerian path e.g. trinity assembly
**** [[http://goo.gl/8Y9nSH][Hamiltonian path]]



* Schools/Areas
Massachussetts
Pennsylvania
North Carolina
New York
Virginia
Maryland
Michigan
South Carolina
Alabama
Georgia
Florida
Arizona
Connecticut
Ohio
Illinois

* Textbooks
** TODO Biological Modeling and Simulation
** TODO General Statistics
** TODO Thermodynamics
** TODO Understanding Bioinformatics
** TODO Systems Bio Voit + Alon
** TODO Linear Algebra / Differential Equations
** TODO Statistics D.O.E.
** TODO Partial Diff Eq.
** TODO Mathematical Biology
* Fields to Study
** TODO Thermodynamics
** TODO Mathematics
** TODO Statistics
** TODO Economics
** TODO Communication
** TODO Teamwork / Management
** TODO Physics
** TODO Entrepeneurship - Startup Owners Manual - Lean start-up udacity.com


