T H I S    I S    A    T O - D O    L I S T
KEEP TRACK OF HOURS!!

Friday BGSA workshop

* No load fee (cut before the door)

* Consumer reports

* Kiplinger





* TODO General To-do

** TODO Analyses
*** TODO Transcriptome Assembly
**** TODO Convert fasta into gtf, diving operons appropriately?
**** TODO Merge new gtf (transcripts) with CDS gtf and sort
**** Assess quality with transrate
***** Graph with R
**** Operon assessment
***** Comparing to former annotation
****** How is the former operon structure organized?
****** Is this format optimal?
****** How can the new data be put in a comparable format?
****** How can that be compared in high throughput?
***** Summary statistics
****** Distributions
******* Transcript length
******* Number of genes per transcript (operon size)
******* IGR lengths
******* UTR sizes (5', 3')
****** Subsets
******* Antisense
******* CDS-less
******* Distributions of these
****** Finding interesting things
******* subsets + distributions
******* Normality and outlier tests on distributions
*** TODO Gene Ontology
**** Methods???
***** First, map GENE_ids back to NCBI:GI -> Uniprot KB ID
***** OR GI -> DAVID
**** Overall
***** General enrichment for differential expression
**** Subsets
***** Factors:
****** By time point
****** By stress
****** By up/down-regulation
***** Clusters
**** Visualizations
***** Pie charts
***** D3 Interaction matrix
****** Sort by clustering
*** TODO Gene expression clustering
**** Preprocessing
***** Data normalization [[http://manuals.bioinformatics.ucr.edu/home/R_BioCondManual#TOC-Clustering-and-Data-Mining-in-R][Scaling]] (DE normalization??)
***** Regularization? (a la DEseq?
**** Feature selection
***** Differential genes only
**** Distance metric
***** Distance matrix selection (e.g. Minkowski measures: Euclidean, Manhattan, Chebyshev OR Mahalanobis**)
****** Useful when searching for EXACT matches between profiles
***** Correlation matrix ***
***** Conditional entropy
**** Linkage
***** Single (nearest)
***** Complete (farthest)
***** Average
***** Centroid (distance between centroid of each cluster)
***** Ward (minimizes ANOVA SSE between the clusters)
**** THEN: CLUSTER [[http://cran.at.r-project.org/web/views/Cluster.html][R portal to clustering]]
**** Considerations
***** Partical or complete clustering? (Do all genes need a cluster?)
**** Techniques
***** Hierarchical clustering (agglomerative)
****** Assumptions:
******* None, well that the criterion for calculating (a)similarity is appropriate
****** Options:
******* Connectivity constraints
****** Pros:
******* Easy to implement
******* Informative
****** Cons:
******* Debate over utility: pergormance is close to random
***** K-means (Partitive; iterative)
****** Assumptions:
******* Clusters are convex
****** K-medians
****** K-medoids (PAM)
****** Spectral clustering
****** Tight clustering (random resampling)
******* Noise tolerant
****** MGKM (higher cost, slight improvement)
****** IGKA (incremental genetic K-means algo.; combines genetic algorithm and K-means)
******* Uses 'Total within cluster variance' as fitness function
******* Incrementally clusters centroids
******* Sensitive to by GA parameters (mutation rate, generation, etc.)
***** Density based clustering
****** Pros/cons
******* Pros
******** No assumptions on convexity
******* Cons
******** Non-deterministic
****** EnDBSCAN/OPTICS
****** Mean-shift (uses KDE)
***** Self organizing maps (Neural nets)
****** SOTA
****** DGSOT
****** GHTSOM
******* Very unique training criteria
******* New nodes are trained only using input which caused the parent node to fire
******* Inputs mapping to leaf neurons at the top of the hierarchy are usually noise!!
***** Biclustering
****** FLOC (George Church??)
****** Possibilistic biclustering
******* H score
***** Fuzzy clustering
****** Fuzzy C-means/ FLAME
******* Differ by weighting; A gene's FCM membership score is proportional to similarity to cluster mean)
******* FLAME determined by weighted similary to K-nearest neighbors and their membership score to cluster

***** Search based
****** Simulated annealing
****** FLOC
****** Expectation maximization
***** Graph theoretic
****** CAST
**** Evaluation
***** fpc
****** Methods:
******* dbscan
******* mergenormals (gaussian mixture, use with centered, regularized)
******* kmeansruns
******* pamk
****** Resampling:
******* Clusterboot
****** prediction.strength (compute optimal number of clusters)
****** (plotcluster) graphing of a clustering
***** clValid (GO and other metrics@!!)
***** clv
***** clusplot (cluster package)
***** pvclust (hierarchical restrapping -> p-values; remember to transpose)

** Circular graph
*** Nodes are GO terms
**** Relationship: total go-term co-occurence
***** Node magnitude 1: number of genes with go-term in subsets with this relationship
****** Direction of relationship = which term has more genes?
***** Node magnitude 2: Total number of occurences of go-term in all subsets
****** Direction of relationship = which term occurs more frequently?
***** Node magnitude 3: Fraction of occurences where terms co-occur in all subsets
****** Direction of relationship = which term tends to co-occur more with its partner
****** NEEDS WORK: inverse of fraction?
***** Node magnitude 4: Total number of co-occurences with all terms
****** Direction of relationship = which terms have the most co-occurences?
***** Node magnitude 5: p-value??
***** More dominant go-terms will have larger area on the circular plot
***** Colored by dominant node (color indicates biological process??)
*** Nodes are Clusters
**** Relationship: Stability metric
**** Relationship: Intracluster variance metric
**** Relationship: Ratio of intracluster to intercluster variance
***** Limit of perfect clustering -> 0
***** Limit of no clustering -> 1
**** Relationship: Ratio of intracluster variance to geometric mean of intercluster variances
*** Nodes are genes
**** Relationship: gene co-occurence
***** Node magnitude 1: Number of clusters that genes co-occur across subsets
***** Node magnitude 2: Number of GO-terms that genes share? (meh)
** Other d3 graphs:
*** Scatterplots
**** MA plots
***** TODO CSV<- gene name, expression value, log fold change, pvalue, and comparison
***** Make index.html and others
**** Correlation matrices
*** Heatmaps
**** With clustering...

** TODO Main graph
*** Add Gene ontology terms
*** Show tooltip
**** Implementation
***** Separate file with gene expression matrix (30 columns x 4k rows)
***** Load file in before hand, plot all points
***** On mouseover:
****** Dim other points
****** Plot 3 lines, each with 10 points
******* [[http://www.d3noob.org/2013/01/smoothing-out-lines-in-d3js.html]['Basis' interpolation]]
****** 4 time points: 15m x2pts, 75m x3pts, 150m x2pts, 270 x3pts
****** Plot [[http://bl.ocks.org/ndarville/6552457][confidence intervals]]
***** On mouseout
***** remove svg? or hide element
** TODO Visualization with d3
*** Differential expression !!!!
**** Scatterplot
***** [[http://www.nytimes.com/interactive/2012/05/17/business/dealbook/how-the-facebook-offering-compares.html?_r=0][Scatterplot with labels]]
***** AND THIS ONE [[http://bl.ocks.org/mbostock/4063663][scatteplot with brushing]]
***** View the gene name, differential expression condition, pvalue others
**** Linkages: GO term with genes
***** [[http://www.findtheconversation.com/concept-map][concept map]]
***** Reorganizing network map [[http://blog.nextgenetics.net/demo/entry0019/demo.html][GO terms]]
***** x DE genes in the center
***** linked to GO terms on the outside
***** Each gene is linked to a condition/time combination? show expression level, percentage changes?
***** Allows more detailed exploration of the gene
***** Each gene ontology is linked to a graph which describes the enrichment of the gene ontology term in certain conditions

**** Interactive heatmap
***** [[http://bl.ocks.org/ianyfchang/8119685][Interactive heatmap]]
***** Ordering by cluster?? [[http://blog.nextgenetics.net/demo/entry0044/][condition comparison]]
***** Display fold change?
*** Coexpression clustering with d3
**** Circular clustering and interaction
***** Demo code: [[https://mbostock.github.io/d3/talk/20111116/bundle.html][Plain html: view source]]
***** BETTER SOURCE: https://gist.github.com/mbostock/1044242
***** Each record of json database has 'imports' a dictionary of linkages
***** Each import becomes colored red when hovering over a particular gene.
***** But, Each record that imports that gene is colored in green
***** USE Projects/d3/hierarchical_edge_bundling_interactive/bundle.html
***** Load data into json database
**** Circular clustering with magnitude
***** May be useful for genes or for gene ontology terms
***** [[http://bost.ocks.org/mike/uberdata/][Circular plot]]
**** Hive plot clusterin and interaction
***** May be harder to use... [[http://bost.ocks.org/mike/hive/][hive plot]]
**** Heatmap: clustering, [correlation magnitude, p-value?] - gene vs GO term
*** Gene ontology visualizations
**** Node link tree: [[http://bl.ocks.org/mbostock/4063550][Reingold-Tilford algorithm]]
**** Circular interaction with magnitudes [[http://bost.ocks.org/mike/uberdata/][Circular interaction]]
**** DITTO: [[http://redotheweb.com/DependencyWheel/][Dependency wheel]]
**** Genes vs gene ontology terms - co-occurence matrix: [[http://bost.ocks.org/mike/miserables/][Les Mis co-occurrence]]
**** 
*** Correlation matrices
**** [[https://mbostock.github.io/d3/talk/20111116/iris-splom.html][interactive subsetting]]
**** For differential expression viewing
**** Another version (older): [[http://benjiec.github.io/scatter-matrix/demo/demo.html][older]]
*** Sankey (non-circular) Flow diagrams with magnitude
**** May be useful for visualizing the change in gene ontology over time...
**** [[http://bost.ocks.org/mike/sankey/][Sankey diagrams]]
**** Can this be modified to display the number associated with the magnitude?
**** As alluvial diagram
*** Sunburst Flow diagrams with percentages
**** Very exhaustive... may visualize something interesting. or not.
**** [[http://bl.ocks.org/kerryrodden/7090426][Sequence sunburst]]
*** Slider subsetter with histograms
**** [[http://square.github.io/crossfilter/][Crossfilter]]
*** Genomic viewer for coverage
**** [[http://square.github.io/crossfilter/][Cubism]]
*** As an illustration to show a eulerian path e.g. trinity assembly
**** [[http://goo.gl/8Y9nSH][Hamiltonian path]]



** Transrate
*** TODO Convert BLAT alignment results to gtf
*** TODO Number of reads per transcript
** cis-software
*** Read Cufflinks GTF file in, create reverse-GTF
*** Set difference of BAM - reverse GTF (filters entries not on opposite strand)
*** Save as is?
*** Pipe to set difference of filtered BAM - CDS? Forward Cufflinks GTF?
*** Pipe to Cufflinks assembly
*** !!!! IDEA !!!!
**** Create a distribution of coverage (likely lognormal) for sense transcription
**** Defined as transcripts with lengths greater than X and with genes mapping back on to them
**** Then use a statistical hypothesis test to filter reads/transcripts with short length and low coverage.
**** i.e. filter the transcript if the majority of its coverage is less than the threshold for sense transcription.
**** (p < 0.05)

** Metrics
*** Number of TSS/CisRNA
*** Distributions of reads/cisRNA
*** Distribution of reads along TSS (same as first item of todo list)
*** Size distribution of cisRNA / UTR region
*** Number of cisRNA that overlap RBS, start codon
*** Comparison of operon organization of Paredes.

** TODO Reach out to Pam, Blake, Cathy, Shawn
*** Hello _____
*** I am wrapping up the first stage of my thesis research under Dr. Papoutsakis and I am looking
*** to form my thesis committee. Given your background in (RNA, small RNA and NGS, NGS and bioinformatics)
*** I believe that you might be interested in my project. I would love to tell you briefly about my research.
*** Thank you for your time.


** NOTE: For TSS annotation, use both non-unique and unique reads to detect TSS
** Cuffdiff contains differential promoter usage information
** TODO Galaxy
*** TODO Go through a worked example
*** TODO Upload data through ftp/filezilla
*** TODO Document the steps required
*** Talk with Shawn about setting up a local Galaxy install
**** Step 1: Install- Karol
**** Step 2: Linking tools: Matt and Karol
**** Step 3: Configuring user accounts/account requests- Karol
**** Step 4: Configuring data libraries/datasets: Shawn/Karol/Matt
**** Step 5: Testing that users can run jobs locally on server node
**** Step 6: Configuring Galaxy with Torque (Watch out for [[http://goo.gl/acKZ4x][Torque]]!) Resource management DRMAA
**** Step 7: [[https://wiki.galaxyproject.org/Admin/Config/Performance/Cluster#Submitting_Jobs_as_the_Real_User][Submitting links as the real user]]
**** 








* Computatxbional
** Data exploration
*** Library complexity
**** Fraction of unique single reads of all mapped reads
**** Fraction of unique pairs of all mapped pairs
*** Coverage plot
**** Coverage vs gene (%, 5'->3') (Express coverage over each gene, where gene length is expressed as percent)
*** 
*** 
*** Correlations between replicates (FPKM)
**** Pearson
**** Spearman
*** Coorelation plot
**** Expression condition A vs expression condition B
**** E.G. x vs. y, y vs. z, x vs. z
**** Log transformed (shows more variation)
**** What are the genes that deviate? How do we find them?
***** Perform regression, calculate the residuals for each gene, sort the list by residuals
*** Strandedness plots
*** [[http://smithlab.usc.edu/plone/software/librarycomplexity][PRESEQ!!!]]
**** Preseq is a library that randomly samples reads from a sequence alignment and uses this to calculate the
**** redundancy of smaller libraries.

** Further exploration
*** ADDRESS OVERPLOTTING BY [[http://gettinggeneticsdone.blogspot.com/2012/07/fix-overplotting-with-colored-contour.html][Colour contour plot (CLICK ME!)]]
*** Circos
**** Histogram
**** Scatterplot
**** Line graph
**** Heatmap
**** Average read count
**** Stack circos png into 3D
***** Sliceomatic (Matlab)
**** Create gif from circos pngs
***** Hist/Line/Scatter
***** Heatmap
*** Volcano plots (MA plot: log2 fold change vs log normalized counts) w/ highlighting significant genes
*** Significance plot (-log10(p-value/FDR) vs log2(fold change)
*** Variation plot: variance (transformed?) vs log normalize counts
** Differential expression
*** Cuffdiff/cuffnorm
*** DEseq
*** MLseq
** Visualize in circos

* NGS tutorial workshop
** PhiX reads
** Ask students to create a galaxy account and load the phiX genome into their data

* Social
** Events
** Personal
*** Valentine's Day
**** Set experiments
**** Get gift
**** Practice ingredients
*** Allison's 40th
**** Find out reservation policy
**** Find out about classes?


* THESIS
** Transcription start site
** New sRNAs
** Clustering by expression pattern(old data)
** Use clusters and TSS info to predict promoter motifs
*** Check if motifs are similar to known.
*** Split clusters into training and test data
*** Use the training data to build a motif
*** Test the motif against the test dataset
*** If the motifs match well, arrive at consensus sequence

** Validate by multiplex 5' RACE?

* Process old data
** Do process for one dataset
*** Quality contorl script BASH
*** Trimming/clipping/filtering/qc script BASH + RUBY
**** Create parsing script to flags files/datasets with poor statistics? From which output?
**** Use flag as input for next round of Trimming/clipping/filtering/qc
*** Mapping BASH
*** Post-mapping quality
** Repeat using recursion for files in a directory
*** Retool scripts for handling these data

* COURSERA
** TODO Read the 'How to Design Programs' Felleisen et al.

* Schools/Areas
Massachussetts
Pennsylvania
North Carolina
New York
Virginia
Maryland
Michigan
South Carolina
Alabama
Georgia
Florida
Arizona
Connecticut
Ohio
Illinois


* Textbooks
** TODO Biological Modeling and Simulation
** TODO General Statistics
** TODO Thermodynamics
** TODO Understanding Bioinformatics
** TODO Systems Bio Voit + Alon
** TODO Linear Algebra / Differential Equations
** TODO Statistics D.O.E.
** TODO Partial Diff Eq.
** TODO Mathematical Biology
* Professional Development
** TODO Claude Shannon - Mathematical Theory of Communication
* Fields to Study
** TODO Thermodynamics
** TODO Mathematics
** TODO Statistics
** TODO Economics
** TODO Communication
** TODO Teamwork / Management
** TODO Physics
** TODO Entrepeneurship - Startup Owners Manual - Lean start-up udacity.com


